keras_hub/src/utils/transformers/export_gemma_to_safetensor.py

```diff
@@ -0,0 +1,200 @@
+        value_kernel = torch.from_numpy(
+            value_kernel
+        ).T  # Shape: [head_dim, hidden_size]
+        weights_dict[f"model.layers.{i}.self_attn.v_proj.weight"] = value_kernel
+
+        # Attention output projection (o_proj)
+        out_kernel = decoder_layer.attention.output_dense.get_weights()[
+            0
+        ]  # Shape: [num_heads, head_dim, hidden_size]
+        out_kernel = (
+            torch.from_numpy(out_kernel)
+            .permute(2, 0, 1)
+            .reshape(backbone.hidden_dim, -1)
+        )  # Reshape to [hidden_size, num_heads * head_dim]
+        weights_dict[f"model.layers.{i}.self_attn.o_proj.weight"] = out_kernel
+
+        # Post-attention normalization (post_attention_layernorm)
+        post_attn_norm = decoder_layer.pre_ffw_norm.get_weights()[
+            0
+        ]  # Shape: [hidden_size]
+        weights_dict[f"model.layers.{i}.post_attention_layernorm.weight"] = (
+            torch.from_numpy(post_attn_norm)
+        )
+
+        # MLP gate projection
+        gate_kernel = decoder_layer.gating_ffw.get_weights()[
+            0
+        ]  # Shape: [hidden_size, intermediate_size]
+        gate_kernel = torch.from_numpy(
+            gate_kernel
+        ).T  # Shape: [intermediate_size, hidden_size]
+        weights_dict[f"model.layers.{i}.mlp.gate_proj.weight"] = gate_kernel
+
+        # MLP up projection
+        up_kernel = decoder_layer.gating_ffw_2.get_weights()[
+            0
+        ]  # Shape: [hidden_size, intermediate_size]
+        up_kernel = torch.from_numpy(
+            up_kernel
+        ).T  # Shape: [intermediate_size, hidden_size]
+        weights_dict[f"model.layers.{i}.mlp.up_proj.weight"] = up_kernel
+
+        # MLP down projection
+        down_kernel = decoder_layer.ffw_linear.get_weights()[
+            0
+        ]  # Shape: [intermediate_size, hidden_size]
+        down_kernel = torch.from_numpy(
+            down_kernel
+        ).T  # Shape: [hidden_size, intermediate_size]
+        weights_dict[f"model.layers.{i}.mlp.down_proj.weight"] = down_kernel
+
+    # Map final normalization
+    final_norm = backbone.get_layer("final_normalization").get_weights()[
+        0
+    ]  # Shape: [hidden_size]
+    weights_dict["model.norm.weight"] = torch.from_numpy(final_norm)
+
+    # Tie lm_head.weight to embedding weights
+    weights_dict["lm_head.weight"] = weights_dict["model.embed_tokens.weight"]
+
+    # Load weights into Hugging Face model
+    print("Loading weights into Hugging Face model...")
+    hf_model.load_state_dict(weights_dict, strict=False)
+
+    # Save tokenizer
+    os.makedirs(path, exist_ok=True)
+    proto = keras_model.preprocessor.tokenizer.proto
+    tokenizer_model_path = os.path.join(path, "tokenizer.model")
+    if isinstance(proto, str):
+        shutil.copyfile(proto, tokenizer_model_path)
+    elif isinstance(proto, bytes):
+        with open(tokenizer_model_path, "wb") as f:
+            f.write(proto)
```

Comment:
Do we need to save the proto file this way? Why can't we directly use the SPM file? See vocabulary.spm here inside assets: https://www.kaggle.com/models/keras/gemma?select=tokenizer.json